source/common/compressor/zlib_compressor_impl.cc

```diff
@@ -68,18 +68,41 @@ void ZlibCompressorImpl::process(Buffer::Instance& output_buffer, int64_t flush_
   }
 
   if (flush_state == Z_SYNC_FLUSH) {
```

Comment:
nit/question: IMO it's a bit strange to finish in the context of a sync_flush operation. They are distinct operations from the deflate perspective. Since I think now when we call flush() we actually want to finish(), should we remove the flush() method for now and just add a direct finish() method?

Reply:
@mattklein123 yes, from my test it works fine to just calling `finish()` from the filter. 

Since the `flush()` would not be used anymore, I think I need to: update the tests as well. Since the current tests are around `Z_SYNC_FLUSH`.

Reply:
@mattklein123 I'm in favor to delete any method that is not being used. @dio It makes sense to replace flush() with finish(), but we still use Z_SYNC_FLUSH internally, correct? FYI, as soon as I get a chance I would like to revisit this entire implementation anyway, so don't worry too much about the design.

Reply:
> It makes sense to replace flush() with finish(), but we still use Z_SYNC_FLUSH internally, correct?

Reply:
> I think without Z_SYNC_FLUSH-ing it is fine. Z_SYNC_FLUSH adds that 0000ffff right before crc32 and isize, which probably it is not necessary?

@dio Good point, but I think we still need sync for byte padding (I'll need to look into it). Here is what others have done. It seems that Z_SYNC_FLUSH is still broadly used.
  
https://github.com/apache/trafficserver/blob/711c4f4be103b44c1971196297c373d7684a0db5/plugins/gzip/gzip.cc#L319

https://github.com/facebook/proxygen/blob/1084545a0015b65882e3256fe18f505942472d40/proxygen/lib/utils/ZlibStreamCompressor.cpp#L146

https://github.com/nginx/nginx/blob/53eae18826d13cb4c7db6680029cac53bfb2a18a/src/http/modules/ngx_http_gzip_filter_module.c#L745

> Fantastic! Probably we can include deflate and deflateRaw as well

I think to do one for Brotli as well would be cool. Some interesting use-cases can be explored when multiple compression types are available :)